src/AzureBlobStorageUpdateSession/UpdateSession/AzureStorageUpdateSessionDiModule.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using Autofac;
+using Etg.Yams.Azure.Lease;
+using Etg.Yams.Azure.UpdateSession.Retry;
+using Etg.Yams.Update;
+using Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling;
+
+namespace Etg.Yams.Azure.UpdateSession
+{
+    public class AzureStorageUpdateSessionDiModule
+    {
+        private readonly IContainer _container;
+        private const string UpdateSessionRetryStrategyModuleName = "updateSessionRetryStrategy";
+
+        public AzureStorageUpdateSessionDiModule(
+            string clusterId,
+            string instanceId,
+            string updateDomain,
+            string connectionString,
+            TimeSpan updateSessionTtl,
+            int storageExceptionRetryCount = 20,
```

Comment:
Do you have to default the parameter values or can you force the caller to always pass them in?

Reply:
they are defaulted here for convenience for people who don't care